src/simularium/LocalFileConnecton.ts

```diff
@@ -0,0 +1,194 @@
+        // not implemented
+    }
+
+    public sendModelDefinition(_model: string): void {
+        // not implemented
+    }
+
+    /**
+     * Simulation Control
+     *
+     * Simulation Run Modes:
+     *  Live : Results are sent as they are calculated
+     *  Pre-Run : All results are evaluated, then sent piecemeal
+     *  Trajectory File: No simulation run, stream a result file piecemeal
+     *
+     */
+    public startRemoteSimPreRun(
+        _timeStep: number,
+        _numTimeSteps: number
+    ): void {
+        // not implemented
+    }
+
+    public startRemoteSimLive(): void {
+        // not implemented
+    }
+
+    public startRemoteTrajectoryPlayback(_fileName: string): Promise<void> {
+        const { spatialData, trajectoryInfo } = this.simulariumFile;
+
+        if (!spatialData) {
+            const newError = new Error(
+                "Simularium files need 'spatialData' array"
+            );
+            return Promise.reject(newError);
+        }
+        spatialData.bundleData.sort(
+            (a: VisDataFrame, b: VisDataFrame): number =>
+                a.frameNumber - b.frameNumber
+        );
+        try {
+            this.onTrajectoryFileInfoArrive(trajectoryInfo);
+        } catch (e) {
+            return Promise.reject(e);
+        }
+        return Promise.resolve();
+    }
+
+    public pauseRemoteSim(): void {
+        window.clearInterval(this.playbackIntervalId);
+        this.playbackIntervalId = 0;
+    }
+
+    public resumeRemoteSim(): void {
+        this.playbackIntervalId = window.setInterval(() => {
+            if (
+                this.currentPlaybackFrameIndex >=
+                this.simulariumFile.spatialData.bundleSize
+            ) {
+                this.currentPlaybackFrameIndex =
+                    this.simulariumFile.spatialData.bundleSize - 1;
+                this.pauseRemoteSim();
+                return;
+            }
+            this.onTrajectoryDataArrive(
+                this.getFrame(this.currentPlaybackFrameIndex)
+            );
+            this.currentPlaybackFrameIndex++;
+        }, 1);
+    }
+
+    public abortRemoteSim(): void {
+        window.clearInterval(this.playbackIntervalId);
+        this.playbackIntervalId = 0;
+        this.currentPlaybackFrameIndex = 0;
+    }
+
+    public requestSingleFrame(startFrameNumber: number): void {
+        this.onTrajectoryDataArrive(this.getFrame(startFrameNumber));
+    }
+
+    public gotoRemoteSimulationTime(timeNanoSeconds: number): void {
+        for (
+            let frame = 0,
+                numFrames = this.simulariumFile.spatialData.bundleData.length;
+            frame < numFrames;
+            frame++
+        ) {
+            const frameTime = this.simulariumFile.spatialData.bundleData[frame]
+                .time;
+            if (timeNanoSeconds < frameTime) {
+                const theFrameNumber = Math.max(frame - 1, 0);
```

Comment:
Yes it was copied from VisData.  In the case of LocalFileSimulator, we have to be able to access a time before it was cached in VisData, because I no longer guarantee that every frame is precached.  I have updated this code to copy the algorithm from VisData now